### PR TITLE
Return error code in request exceptions and ignore 438/439 errors (LG-4137)

### PR DESCRIPTION
**Pull request by mitchellhenke**

### CODE REVIEW

File: lib/identity_doc_auth/acuant/request.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 module IdentityDocAuth
   module Acuant
+    # https://documentation.help/AssureID-Connect/Error%20Codes.html
+    # 438 and 439 are frequent errors that we do not want to be notified of
+    IGNORED_ERROR_CODES = [438, 439]
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/request.rb]: let's make it a set?

```suggestion
    IGNORED_ERROR_CODES = Set[438, 439]
```

**Comment by amathews-fs** [comment]
@mitchellhenke remember to bump the version.

File: lib/identity_doc_auth/acuant/request.rb [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+    class RequestError < StandardError
+      attr_reader :error_code
+      def initialize(message, error_code)
+        @error_code = error_code
+        super(message)
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/request.rb]: Do we want to interpolate the error code into the message so we can make it easier to find in NewRelic? I don't think ivars get automatically reported?

```suggestion
      def initialize(message, error_code)
        @error_code = error_code
        super("#{error_code}: #{message}")
```

Comment (zachmargolis) [reply lib/identity_doc_auth/acuant/request.rb]: Also WDYT of moving this RequestError class up a namespace so we can just use the same class in Acuant/LN?

**Review by zachmargolis — APPROVED** [review]
LGTM

File: lib/identity_doc_auth/acuant/request.rb (hunk 2 above)
Comment (mitchellhenke) [reply lib/identity_doc_auth/acuant/request.rb]: could refactor it, but the error code is in the message: https://github.com/18F/identity-doc-auth/blob/main/lib/identity_doc_auth/acuant/request.rb#L94-L98

Comment (mitchellhenke) [reply lib/identity_doc_auth/acuant/request.rb]: Agree on moving it up a namespace

Comment (zachmargolis) [reply lib/identity_doc_auth/acuant/request.rb]: > could refactor it, but the error code is in the message: https://github.com/18F/identity-doc-auth/blob/main/lib/identity_doc_auth/acuant/request.rb#L94-L98

ah perfect, thanks!

**Comment by amathews-fs** [comment]
It seems to be outside of the lines I can specifically call out but can you move the `conn.adapter :net_http` lines in both of the Request classes down to the end of their respective declaration blocks? 

I can't find the documentation now but I believe that the `adapter` has to be the last thing declared in the connection block.

**Comment by mitchellhenke** [comment]
@amathews-fs moved adapter in 951aedc

**Pull request merged**